test/fluree/db/branch_test.clj

```diff
@@ -0,0 +1,280 @@
+(ns fluree.db.branch-test
+  (:require [babashka.fs :refer [with-temp-dir]]
+            [clojure.test :refer [deftest is testing]]
+            [fluree.db.api :as fluree]
+            [fluree.db.test-utils :as test-utils]
+            [fluree.db.util :as util]))
+
+(deftest ^:integration branch-basic-operations-test
+  (testing "Basic branch operations with memory connection"
+    (let [conn @(fluree/connect-memory {})]
+      (try
+        ;; Create ledger and add initial data
+        (testing "Create ledger with initial data"
+          @(fluree/create conn "test-ledger" {})
+          @(fluree/insert! conn "test-ledger"
+                           {"@context" {"ex" "http://example.org/"}
+                            "@graph" [{"@id" "ex:alice"
+                                       "@type" "Person"
+                                       "ex:name" "Alice"}]}))
+
+        ;; Query main branch
+        (testing "Verify initial state in main branch"
+          (let [main-db @(fluree/load conn "test-ledger")
+                results @(fluree/query main-db {"@context" {"ex" "http://example.org/"}
+                                                "select" "?name"
+                                                "where" {"@id" "ex:alice" "ex:name" "?name"}})]
+            (is (= ["Alice"] results) "Should find Alice in main branch")))
+
+        (testing "Create feature branch from main"
+          (let [branch-result @(fluree/create-branch! conn "test-ledger:feature-branch" "test-ledger:main")]
+            (is (= "feature-branch" (:name branch-result)))
+            (is (= "main" (get-in branch-result [:created-from :branch])))
+            (is (some? (:created-at branch-result)))
```

Comment:
We can use `with-redefs` to make `current-time-iso` deterministic, I prefer that approach to just checking that it's present.